cmd/jaeger/internal/extension/jaegerstorage/factoryadapter/factory.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package factoryadapter
+
+import (
+	"context"
+	"io"
+
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	storage_v1 "github.com/jaegertracing/jaeger/storage"
+	"github.com/jaegertracing/jaeger/storage_v2/spanstore"
+)
+
+type Factory struct {
+	logger *zap.Logger
+	ss     storage_v1.Factory
+}
+
+func NewFactory(logger *zap.Logger, ss storage_v1.Factory) spanstore.Factory {
+	return &Factory{
+		logger: logger,
```

Comment:
are you sure you need the logger in the adapter?

Reply:
oh, I overlooked it. I thought the Initialize function was used.